modules/pinger/src/main/java/org/hawkular/component/pinger/MetricPublisher.java

```diff
@@ -47,72 +50,110 @@
+                + "/metrics/numeric/data");
+        // HttpPost request = new HttpPost(configuration.getMetricsBaseUri() + "/metrics/numeric/data");
+        // request.addHeader("tenantId", status.getDestination().getTenantId());
+        // TODO end replace for metrics 0.3.3
+
+        request.setEntity(new StringEntity(payload, ContentType.APPLICATION_JSON));
 
         try {
             HttpResponse response = client.execute(request);
-            if (response.getStatusLine().getStatusCode()>399) {
+            if (response.getStatusLine().getStatusCode() > 399) {
                 Log.LOG.metricPostStatus(response.getStatusLine().toString());
             }
         } catch (IOException e) {
-            e.printStackTrace();  // TODO: Customise this generated block
+            e.printStackTrace(); // TODO: Customise this generated block
         }
     }
 
     /**
-     * Put a list of metric data on the Metrics topic.
-     * @param tenantId Id of the tenant
-     * @param metrics Metrics to publish
+     * Serializes the given {@link PingStatus} and then submits it on the Metrics topic of the bus.
+     *
+     * @param status
+     *            the {@link PingStatus} to publish
      */
     @Asynchronous
-    public void publishToTopic(String tenantId, List<SingleMetric> metrics) {
+    public void publishToTopic(PingStatus status) {
         if (topic != null) {
 
-            try ( ConnectionContextFactory factory = new ConnectionContextFactory(connectionFactory)) {
+            try (ConnectionContextFactory factory = new ConnectionContextFactory(connectionFactory)) {
+
+                List<SingleMetric> singleMetrics = new ArrayList<>();
+
+                final PingDestination dest = status.getDestination();
+                final String resourceId = dest.getResourceId();
+                final long timestamp = status.getTimestamp();
 
-                Map<String,Object> outer = new HashMap<>(1);
+                SingleMetric singleMetric = new SingleMetric(resourceId + ".status.duration", timestamp,
+                        (double) status.getDuration());
+                singleMetrics.add(singleMetric);
+                singleMetric = new SingleMetric(resourceId + ".status.code", timestamp, (double) status.getCode());
+                singleMetrics.add(singleMetric);
 
-                Map<String,Object> data = new HashMap<>(2);
-                data.put("tenantId",tenantId);
-                data.put("data",metrics);
+                Map<String, Object> outer = new HashMap<>(1);
 
-                outer.put("metricData",data);
+                Map<String, Object> data = new HashMap<>(2);
+                data.put("tenantId", status.getDestination().getTenantId());
+                data.put("data", singleMetrics);
 
-                Endpoint endpoint = new Endpoint(Endpoint.Type.TOPIC,topic.getTopicName());
+                outer.put("metricData", data);
```

Comment:
Yes, the code here is not very clean. I have no idea if it works at all. I have just moved it here from other place. Putting the data together like this is not safe. It would be ideal if Metrics API offered some type-safe way to do this, like Inventory does. Maybe it does, I have not investigated. Anyway changing anything on writing to Metrics is not the objective of this PR.